app/transactions/admin/enrollments/update_transaction.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+module Admin
+  module Enrollments
+    # update enrolment after a dateTimes validation
+    class UpdateTransaction # rubocop:disable Metrics/ClassLength
+      include Dry::Transaction
+
+      around :rollback_if_failure
+
+      tee :convert_params_in_datetime
+      step :check_if_the_duration_is_positive
+      step :check_if_datetimes_of_enrollment_are_inside_the_mission_s_period
+      step :check_if_enrollment_are_matching_the_mission_s_timeslots
+      step :check_slot_availability_for_regulated_mission
+      step :check_cash_register_mastery
+      step :update_enrollment
+
+      private
+
+      def rollback_if_failure(input, &block)
+        result = nil
+
+        Enrollment.transaction do
+          result = block.call(Success(input))
+          raise ActiveRecord::Rollback if result.failure?
+
+          result
+        end
+        result
+      end
+
+      def convert_params_in_datetime(input)
+        start_time = convert_datetime(input[:params], 'start_time')
+        end_time = convert_datetime(input[:params], 'end_time')
+        input[:params].merge!(start_time: start_time, end_time: end_time)
+        Success(input)
+      end
+
+      def check_if_the_duration_is_positive(input)
+        failure_message = I18n.t('activerecord.errors.models.enrollment.negative_duration')
+        return Failure(failure_message) unless input[:params][:start_time] < input[:params][:end_time]
+
+        Success(input)
+      end
+
+      def check_if_datetimes_of_enrollment_are_inside_the_mission_s_period(input)
+        failure_message = I18n.t('activerecord.errors.models.enrollment.inconsistent_datetimes')
+        return Failure(failure_message) unless are_inside_the_mission_s_period?(input)
+
+        Success(input)
+      end
+
+      def check_if_enrollment_are_matching_the_mission_s_timeslots(input)
+        return Success(input) unless input[:enrollment].mission.regulated?
+
+        failure_message = I18n.t('activerecord.errors.models.enrollment.time_slot_mismatch')
+        return Failure(failure_message) unless match_a_time_slot?(input)
+
+        Success(input)
+      end
+
+      def check_slot_availability_for_regulated_mission(input)
+        mission = input[:enrollment].mission
+        failure_message = I18n.t('activerecord.errors.models.enrollment.slot_unavailability')
+        return Success(input) if mission.genre != 'regulated'
+
+        return Failure(failure_message) unless are_all_time_slots_available?(input)
+
+        Success(input)
+      end
+
+      def check_cash_register_mastery(input)
+        return Success(input) unless input[:enrollment].mission.regulated?
+
+        failure_message = I18n.t('activerecord.errors.models.enrollment.insufficient_cash_register_mastery')
+        return Failure(failure_message) unless inspect_all_time_slots_for_register_cash_mastery(input)
+
+        Success(input)
+      end
```

Comment:
Since this is duplicated from the related `create_transaction`, and this actually checks if an `Enrollment` is valid, you should extract these methods and add them as validation steps into the `Entrollment` model